src/fuzzing/fuzzrng.rs

```diff
@@ -0,0 +1,105 @@
+use rand::prelude::*;
+use rand::Error;
+
+use random_lfsr_256_galois::{InitRegisterPayload, LFSRGalois, LFSRGaloisBuilder};
+
+// A pseudo RNG which uses a slice for all entropy.  It iterates a given slice
+// many times, scrambling the bits through an LFSR in subsequent passes so that
+// the fuzzer maintains a relationship between the original input bits and the
+// output with some predictability.
+//
+// This differs from BufRng in that it's not intended to be exhaustible since
+// it's being used to generate extensible data structures.
+pub struct FuzzPseudoRng<'slice> {
+    lfsr: LFSRGalois,
+    slice: &'slice [u8],
+
+    // Set on second or subsequent run
+    lfsr_scramble: bool,
+    progress: usize,
+}
+
+impl<'slice> FuzzPseudoRng<'slice> {
+    pub fn new(slice: &'slice [u8]) -> Self {
+        // Ensure the lfsr state is consistent so the entropy bits produce
+        // an identical randomness every time.
+        let lfsr = LFSRGaloisBuilder::new()
+            .set_initial_payload(InitRegisterPayload::Meander)
+            .build();
+        return FuzzPseudoRng {
+            lfsr: lfsr,
```

Comment:
I think it would be good to fix these clippy warnings too

Reply:
a clippy update was issued literally 3 days ago and this pr is a month old.

Reply:
I see. I noticed that some of the new clippy warnings actually requires bumping edition to 2021 to as well

Reply:
hmm good point i'll do that.